src/rulesets/mergers/exceptions.ts

```diff
@@ -0,0 +1,78 @@
+import { extractPointerFromRef, extractSourceFromRef, pointerToPath } from '@stoplight/json';
+import { isAbsolute, join, normalize as pathNormalize } from '@stoplight/path';
+import { RulesetExceptionCollection } from '../../types/ruleset';
+
+const normalize = ($ref: string, rulesetUri?: string): string => {
+  const source = extractSourceFromRef($ref);
+
+  if (typeof source !== 'string') {
+    throw new Error(buildInvalidUriErrorMessage($ref, rulesetUri, 'Missing source'));
+  }
+
+  if (rulesetUri === undefined && !isAbsolute(source)) {
+    throw new Error(
+      buildInvalidUriErrorMessage(
+        $ref,
+        rulesetUri,
+        'Only absolute Uris are allowed when no base ruleset uri has been provided',
+      ),
+    );
+  }
+
+  const pointer = extractPointerFromRef($ref);
+
+  if (typeof pointer !== 'string') {
+    throw new Error(buildInvalidUriErrorMessage($ref, rulesetUri, 'Missing pointer fragment'));
+  }
+
+  try {
+    pointerToPath(pointer);
+  } catch {
+    throw new Error(buildInvalidUriErrorMessage($ref, rulesetUri));
```

Comment:
How about having a dedicated error, such as `InvalidUriError`? It'd be somewhat similar to ValidationError we have in case of rulesets.

Reply:
@P0lip Fixed